api-js/src/auth/check-domain-permission.js

```diff
@@ -0,0 +1,24 @@
+const checkDomainPermission = async (userId, domainId, query) => {
+  let userAffiliatedClaims
+
+  // Retrieve user affiliations and affiliated organizations owning provided domain
+  try {
+    userAffiliatedClaims = await query`
+      LET userAffiliations = (FOR v, e IN 1..1 ANY ${userId} affiliations RETURN e._from)
+      LET domainClaims = (FOR v, e IN 1..1 ANY ${domainId} claims RETURN e._from)
+      LET affiliatedClaims = INTERSECTION(userAffiliations, domainClaims)
+        RETURN affiliatedClaims
+    `
+  } catch (err) {
+    console.error(
+      `Error when retrieving affiliated organization claims for user with ID ${userId} and domain with ID ${domainId}: ${err}`,
+    )
+    throw new Error('Authentication error. Please sign in again.')
+  }
+  const claim = await userAffiliatedClaims.next()
```

Comment:
Should throw this into a try catch just in case something goes wrong and an error is thrown